### ISSUE TITLE
Namespacing and configuring controllers for SiteTree subclasses

### ISSUE THREAD

**Issue by colintucker** [silverstripe/silverstripe-cms#1721]
Currently, with the move to PSR-2 compliant code in v4, `Controller` subclasses should be separated into their own files, instead of being in the same file as their linked `SiteTree` subclasses.

The namespace changes also introduced in v4 allow for these controllers to exist in a different namespace from the model, however, currently they must exist in the same namespace in order for `SiteTree::getControllerName()` [to be able to locate them](https://github.com/silverstripe/silverstripe-cms/blob/3f53da050cecd47605575b4416abfc7fbe4f893c/code/Model/SiteTree.php#L2739-L2751).

This method could be overridden by subclasses to return a controller name in a different namespace, but it feels hacky, and would probably be better solved by adding a configuration option to `SiteTree` such as (courtesy @sminnee ):

```php
use App\Controllers\PageController;

class Page {
  private static $frontend_controller = PageController::class;
}
```

...and then modifying `SiteTree::getControllerName()` to look for this configuration option, with a fallback to the current `{$Page}Controller` and `{$Page}_Controller` (deprecated) functionality.

Thoughts?

**Comment by robbieaverill** [comment]
Duplicated by silverstripe/silverstripe-framework#6549 (sorry, hadn't seen this!) and fixed in #1722

**Comment by chillu** [comment]
Closing as duplicate
